contracts/ScheduledPayment.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.9;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+
+contract ScheduledPayment is Module {
+    event ScheduledPaymentSetup(
+        address indexed initiator,
+        address indexed owner,
+        address indexed avatar,
+        address target,
+        address config
+    );
+    event PaymentScheduled(uint256 nonce, bytes32 spHash);
+
+    event ConfigSet(address config);
+
+    address public config;
+    uint256 public nonce;
+    // Mapping of nonce to SP hash.
+    mapping(uint256 => bytes32) public spHash;
```

Comment:
What's the meaning of nonce here? Would it be easier to use an `EnumerableSet` for scheduled payment hashes?

Reply:
Yes, I think it would be easier.

Reply:
updated